eZ/Bundle/EzPublishCoreBundle/EventListener/LocaleListener.php

```diff
@@ -47,15 +47,17 @@ public function setLocaleConverter(LocaleConverterInterface $localeConverter)
     public function onKernelRequest(GetResponseEvent $event)
     {
         $request = $event->getRequest();
+
         if (!$request->attributes->has('_locale')) {
+            $convertedLocales = [];
             foreach ($this->configResolver->getParameter('languages') as $locale) {
                 $convertedLocale = $this->localeConverter->convertToPOSIX($locale);
                 if ($convertedLocale !== null) {
-                    // Setting the converted locale to the _locale request attribute, so that it can be properly processed by parent listener.
-                    $request->attributes->set('_locale', $convertedLocale);
-                    break;
+                    $convertedLocales[] = $convertedLocale;
                 }
             }
+
+            $request->attributes->set('_locale', $request->getPreferredLanguage($convertedLocales));
```

Comment:
Ah, I see now why @alongosz said it's a break here. This will affect any kind of translated strings, also for front end. It afaik means any response varies by user preferred languages if we do this change.

Maybe we need input from @bdunogier here on suggestions for alternative ways to deal with this.

Reply:
What if we would make it configurable to maintain current and user preferred behaviour?

Reply:
Could be a flag we set internally on some "special" requests, but don't think I would opt for a user configurable setting for this.

Reply:
Or do you have a feature request for admin ui in mind here as well? (like language selector)

Reply:
By user I meant, system integrator for now, as platform UI in 1.x relays on user browser language. On the roadmap for 2.x we have user preferences. So despite the configuration in siteaccess, they can still set preferred language for UI in admin perferences. That setting will be persisted and as long as user is logged in, he will get UI in preferred language.

Reply:
@andrerom yes, you are totally right, it will affect *every* string which is translated by Symfony's translator.

Reply:
@lserwatka Clear. Well my thinking for admin v2 was that a symfony event  or similar would inject the preference, 1. if set, 2. only when you run on admin.
- Inject to symfony translation kind of like here but with conditions above
- Inject to Site [Access Aware] Repository, this is made possible in #2064 where it is referred to as `context language`, and it will thus become the preferred language over config languages over main language, unless API consumer specifies languages explicitly